app/src/main/java/org/mozilla/focus/session/ui/TabViewHolder.kt

```diff
@@ -6,40 +6,39 @@ package org.mozilla.focus.session.ui
 
 import android.animation.Animator
 import android.animation.AnimatorListenerAdapter
-import androidx.recyclerview.widget.RecyclerView
 import android.view.View
 import android.widget.TextView
-import mozilla.components.browser.session.Session
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.browser.state.state.TabSessionState
 import org.mozilla.focus.R
 import org.mozilla.focus.ext.beautifyUrl
 import org.mozilla.focus.ext.requireComponents
 import org.mozilla.focus.telemetry.TelemetryWrapper
 import java.lang.ref.WeakReference
 
-class SessionViewHolder internal constructor(
-    private val fragment: SessionsSheetFragment,
+class TabViewHolder internal constructor(
+    private val fragment: TabSheetFragment,
     private val textView: TextView
 ) : RecyclerView.ViewHolder(textView), View.OnClickListener {
     companion object {
-        @JvmField
-        internal val LAYOUT_ID = R.layout.item_session
+        internal const val LAYOUT_ID = R.layout.item_session
     }
 
-    private var sessionReference: WeakReference<Session> = WeakReference<Session>(null)
+    private var tabReference: WeakReference<TabSessionState> = WeakReference<TabSessionState>(null)
```

Comment:
nit: TabSessionState probably doesn't need to be a weakref any more.

Reply:
Oh, yeah, right. I didn't even process that code. Looks like we only keep the tab around to select it later, and we can do that with just the tab ID now, so no need to keep more than that. :)